cmd/cloud-gate/main.go

```diff
@@ -19,30 +19,35 @@ var (
 		time.Minute*5, "Configuration check interval")
 	configurationUrl = flag.String("configurationUrl",
 		"file:///etc/cloud-gate/conf.yml", "URL containing configuration")
-	portNum = flag.Uint("portNum", 443,
+	portNum = flag.Uint("portNum", 4443,
```

Comment:
I don't agree with this change. The default should be correct for Production use.

Reply:
Yes, the default should be correct for production use, this is why this change is needed.

Current code is broken because `443` is a privileged port and the program exits with error if it is run as root.

Reply:
OK, just remove the EUID check and restore the default port number to 443.

Reply:
Running service such as `cloud-gate` with root privilege is dangerous and is bad practice. There are several better alternatives we can use to run the service on privileged port without granting it root access:

1. Reverse proxy.
    This is arguably the most widely used approach at the price of some deployment complication. With this setup a well established reverse proxy (such as Nginx) runs on privileged port and routes requests to `cloud-gate` which runs on unprivileged port.

2. SystemD socket activation.
    We can rely on SystemD to bind to the privileged port and create a socket that can be used by Go program.

3. Linux Capabilities (setcap)
    This approach utilizes POSIX Capabilities feature to enable certain capabilities for a process without giving it full root privileges. 
    See [this question](https://superuser.com/questions/710253/allow-non-root-process-to-bind-to-port-80-and-443)  for details.
    This should be done on deployment step with:
     ```
    sudo setcap CAP_NET_BIND_SERVICE=+eip /path/to/binary 
    ```

This particular pull request is now outdated, so one of these options should be implemented in subsequent pull requests. I believe for this particular use case option 3 may be the best.